cli/derive/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+//! Crate with a proc macro for torii endpoint generation
+
+use proc_macro::TokenStream;
+use proc_macro2::Span;
+use quote::quote;
+use syn::{
+    parse::{Parse, ParseStream},
+    parse_macro_input,
+    punctuated::Punctuated,
+    Ident, LitInt, LitStr, Result as SynResult, Token,
+};
+
+/// Generate warp filters for endpoints, accepting functions
+/// with any positive number of arguments within the range of `u8`.
+///
+/// Only the endpoint functions stated explicitly in the macro invocation
+/// are created.
+///
+/// There are two kinds of accepted arguments. One is supplying
+/// an integer literal denoting the number of arguments in a function
+/// that the endpoint accepts. The endpoint name is generated automatically
+/// in this case and will be in the shape of `endpoint{arg_count}`.
+///
+/// Another kind is a colon-separated string literal
+/// followed by an integer literal, denoting custom name of the endpoint being
+/// created and the number of arguments in a function that it accepts.
+///
+/// Also relies on `WarpResult` custom wrapper,
+/// and thus any module using this macro should also reexport
+/// the former.  
+///
+/// # Panics:
+/// 1) When provided with neither a string nor integer literal.
+/// 2) When any of the argument count literals are not unique.
+/// 3) When the colon-separated form has spaces in the provided name.
+#[proc_macro]
+pub fn generate_endpoints(input: TokenStream) -> TokenStream {
```

Comment:
I'd add the usage examples, especially for the second synax, as it's not used anywhere

Reply:
it's used, but in one place only. Yeah, example in the docs or a compile test or a runtime test. But because this is used in only one place and I don't think it will spread around, I'm ok with it as is